### PR TITLE
Only allow blocks with non empty transactions set

### PR DESCRIPTION
**Pull request by TheMMaciek**

### CODE REVIEW

File: modules/dag-l1/src/main/scala/org/tessellation/dag/l1/domain/consensus/block/BlockConsensusCell.scala [hunk 1]
```diff
@@ -286,29 +286,37 @@ object BlockConsensusCell {
...
-                      owner = roundData.owner,
-                      CreatedInvalidBlock
+            roundData.formBlock() match {
+              case Some(block) =>
+                Signed.forAsyncKryo(block, ctx.keyPair).flatMap { signedBlock =>
+                  ctx.blockValidator
+                    .validate(signedBlock, validationParams)
+                    .flatTap { validationResult =>
+                      if (validationResult.isInvalid) Logger[F].debug(s"Created block is invalid: $validationResult")
+                      else Applicative[F].unit
```
Comment (marcinwadon) [inline modules/dag-l1/src/main/scala/org/tessellation/dag/l1/domain/consensus/block/BlockConsensusCell.scala:296]: as written on slack, you can simplify this using applicative whenA

**Review by marcinwadon — APPROVED** [review]

**Review by tbekas — APPROVED** [review]
Minor comment, otherwise looks good.

File: modules/dag-l1/src/main/scala/org/tessellation/dag/l1/domain/consensus/block/BlockConsensusCell.scala [hunk 2]
```diff
@@ -286,29 +286,37 @@ object BlockConsensusCell {
...
+                    }
```
Comment (kpudlik) [inline modules/dag-l1/src/main/scala/org/tessellation/dag/l1/domain/consensus/block/BlockConsensusCell.scala:297]: eventually

```scala
.flatTap { validationResult =>
  Applicative[F].whenA(validationResult.isInvalid) {
    Logger[F].debug(s"Created block is invalid: $validationResult")
  }
}
```

**Review by kpudlik — APPROVED** [review]

File: modules/shared/src/main/scala/org/tessellation/ext/codecs.scala [hunk 3]
```diff
@@ -30,4 +34,11 @@ object codecs {
       }
   }
 
+  object NonEmptySetCodec {
+
+    def decoder[A: Decoder: Order: Ordering]: Decoder[NonEmptySet[A]] =
```
Comment (tbekas) [inline modules/shared/src/main/scala/org/tessellation/ext/codecs.scala:39]: You can require only `Odering` and create `Order` necessary for `decodeNonEmptySet` using the `Ordering`.

**Review by tbekas — APPROVED** [review]
LGTM

**Review by tbekas — APPROVED** [review]
LGTM

**Pull request merged**